### DISCUSSION TITLE
Features

### DISCUSSION THREAD

**Discussion by Iandiehard** [Iandiehard/diag-client-lib#25] [Ideas]
Hello Everyone,

If you need any special features to be included in the next release of diag-client-lib. Please comment in this discussion.

I will include your requested feature in the next releases 

Regards.
AD

**Comment by Alexpux** [comment]
Hi! Firsts thank for your work!!!
Probably its good idea to add to client app ability to discovery gateways IP addresses in network and use them instead passing from config file. What do you think?

**Comment by Iandiehard** [comment]
Hello @Alexpux ,

Thanks for trying out diag-client-lib :)

If you are talking about the ability to discover available Diag servers in a network, its already supported by DiagClientLib. So, DiagClientLib
can broadcast Vehicle Identification request to the whole network and provides the caller a list of available DiagServer.

Here is the Api to send Vehicle Identification request:-
https://github.com/Iandiehard/diag-client-lib/blob/e25568b151674bad3e168cb3f56bab7adea1b3ca/diag-client-lib/appl/include/diagnostic_client.h#L87

Also just for the information:-

The Ip address you see in the config file for each conversation is basically the local ip address that is used as source Ip address for sending tcp data https://github.com/Iandiehard/diag-client-lib/blob/e25568b151674bad3e168cb3f56bab7adea1b3ca/diag-client-lib/appl/etc/diag_client_config.json#L15

Please let me know if you have any question.

Thanks,
AD

**Reply by Alexpux** [reply]
Thanks for info, seems I some misunderstand config file :) Looking to your great work!!! I will try to work more with your lib and give feedback.

**Reply by Iandiehard** [reply]
Great, I then wait for your feedback. If you feel the Api documentation are not self explanatory , please let me know. I will improve it to my best.
